### ISSUE TITLE
USB current power for charging devies and phones

### ISSUE THREAD

**Issue by bbriton** [cmer/gigabyte-z390-aorus-master-hackintosh#69]
I have the same ports open as CMER exactly and had it all running perfect with clover.  When I switched to Open Core I couldn't get the USB ports to charge at the high current rate as per before. (about 1% battery charge per min on my Iphone X) I had done the USBMap before in CLover and used the same one I made and also tried the one from this site as I use same ports but would not work.  I noticed in Hackintool that the USB ports listed were more than supposed to be and not correct.  I rebuilt the USBMap with hackintool and also replaced SSDT-EC-USBX.aml with the one produced from Hackintool and that fixed it.  All works 100% now and am on OC0.5.9.
I believe the SSDT-EC-USBX.aml provided in the OpenCore package was causing the issue. I have attached the USBMap and SSDT-EC-USBX files I made that work now for your pleasure.
[Archive.zip](https://github.com/cmer/gigabyte-z390-aorus-master-hackintosh/files/4729946/Archive.zip)

**Comment by riley21** [comment]
@bbriton followed your steps and got usb current power working again.  Before my ipad said "not charging" when plugged in and now they indicate they are charging.  Note that when you export files with Hackintool it named the kext USBPorts.kext. (i didn't have excess ports listed so I didn't have to redo the mapping) I did have to rename the kext to USBMap.kext before sticking it back into the EFI folder

**Comment by bbriton** [comment]
Glad it worked for you. Yes I should have mentioned Hackintool names the file different. I too renamed the file like you did.

**Comment by bbriton** [comment]
I only charge with the USB 3.0 ports since as per the below text I copied from the Gigabyte site for our Master motherboard:

USB TurboCharger
GIGABYTE patent pending USB TurboCharger technology allows users to recharge their mobile devices with stunning speed, up to 50% in less than 30 minutes*. By connecting the Front USB 3.0 Type-A chassis cable to our motherboard, users can utilize the USB TurboCharger to recharge both Android devices with QC 3.0 and Apple devices with Apple Fast-Charge function.

**Comment by riley21** [comment]
@bbriton should I be resetting my NVRAM when I modify the SSDT-EC-USBX.aml & USBMap.kext?

**Comment by bbriton** [comment]
I don’t believe I had to reset my NVRam for this. But I did reset it a few times prior. I am no expert just a tester. Maybe a true expert can comment if necessary.
